specification/ai/OpenAI.Assistants/main.tsp

```diff
@@ -0,0 +1,46 @@
+import "@azure-tools/typespec-azure-core";
+import "@typespec/http";
+import "@typespec/versioning";
+import "./assistants/main.tsp";
+import "./threads/main.tsp";
+import "./messages/main.tsp";
+import "./runs/main.tsp";
+import "./run_steps/main.tsp";
+import "./files/main.tsp";
+
+using TypeSpec.Http;
+using TypeSpec.Versioning;
+
+using Azure.Core;
+
+#suppress "@azure-tools/typespec-azure-core/casing-style" "OpenAI is a case-sensitive name"
+@useAuth(
+  ApiKeyAuth<ApiKeyLocation.header, "api-key"> | OAuth2Auth<[
+    {
+      type: OAuth2FlowType.implicit,
+      authorizationUrl: "https://login.microsoftonline.com/common/oauth2/v2.0/authorize",
+      scopes: ["https://cognitiveservices.azure.com/.default"],
+    }
+  ]>
+)
+@service({
+  title: "Azure OpenAI",
+})
+@server(
+  "{endpoint}/openai",
```

Comment:
I have play around with OpenAI which has url = https://api.openai.com/v1/{assistants|threads|file|...} that has no `/openai` keyword added to the endpoint. If we can change this line to `{endpoint}` only without hard-coded `/openai` will make this implementation more flexible. Also It will make this implementation works for both Azure and Non Azure endpoint.

We can add `/openai`  when setting the value of {endpoint} = azureEndpoint+"/openai", that will make Azure OpenAI works as well.

Reply:
Great point! I'll remove the /openai.